tests/launch.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(feature = "openssl")]
+
+mod common;
+
+use sev::certs::Chain;
+use sev::firmware::Firmware;
+use sev::launch::{HeaderFlags, Launcher, Policy};
+use sev::session::Session;
+
+use sev::Generation;
+
+use codicon::Decoder;
+use kvm_bindings::kvm_userspace_memory_region;
+use kvm_ioctls::{Kvm, VcpuExit};
+use mmarinus::{perms, Kind, Map};
+use serial_test::serial;
+
+use std::convert::TryFrom;
+use std::fs::File;
+
+const CODE: &[u8] = &[
+    0xf4, // hlt
+];
+
+fn __get_cert_chain(sev: &mut Firmware) -> Chain {
+    let mut platform = sev.pdh_cert_export().unwrap();
+
+    let id = sev.get_identifer().unwrap();
+    let url = format!("https://kdsintf.amd.com/cek/id/{}", id);
+
+    let mut resp = reqwest::blocking::get(&url)
+        .unwrap()
+        .error_for_status()
+        .unwrap();
+
+    let mut cek = vec![];
+    let _ = resp.copy_to(&mut cek).unwrap();
+    platform.cek = sev::certs::sev::Certificate::decode(&mut &cek[..], ()).unwrap();
+
+    let ca = Generation::try_from(&platform).unwrap().into();
+
+    Chain { sev: platform, ca }
+}
+
+fn get_cert_chain(sev: &mut Firmware) -> Chain {
+    let cached_chain = common::cached_chain_path().unwrap();
+
+    File::open(cached_chain.clone())
+        .and_then(|mut f| Chain::decode(&mut f, ()))
+        .unwrap_or_else(|_| {
+            use codicon::Encoder;
+
+            let chain = __get_cert_chain(sev);
+            let directories = cached_chain.parent().unwrap();
+            std::fs::create_dir_all(directories).unwrap();
+            let mut f = File::create(cached_chain).unwrap();
```

Comment:
Hmm, that is racy.. One should create a tmpfile first with a unique name in that directory, then rename, when it was written completely, otherwise other processes would read an incomplete file.

Reply:
Done!

Reply:
@haraldh There should really be a crate for that...